pandas/core/dtypes/dtypes.py

```diff
@@ -963,6 +1016,10 @@ def __eq__(self, other):
             from pandas.core.dtypes.common import is_dtype_equal
             return is_dtype_equal(self.subtype, other.subtype)
 
+    def __setstate__(self, state):
+        # for pickle compat.
```

Comment:
i think would add a bit more comment here. (in all of the setstates to indicate why this is needed, maybe reference this issue)

Reply:
OK, expanded the comment!